src/middlewared/middlewared/plugins/netdata.py

```diff
@@ -0,0 +1,208 @@
+                        ip for ip in await self.middleware.call('interfaces.ip_in_use') if ip['address'] == bind_to_ip
+                    ]
+            ):
+                verrors.add(
+                    'netdata_update.bind_to',
+                    'Please provide a valid bind to ip'
+                )
+        else:
+            verrors.add(
+                'netdata_update.bind_to',
+                'This field is required'
+            )
+
+        update_alarms = data.pop('update_alarms')
+        valid_alarms = await self.list_alarms()
+        if update_alarms:
+            for alarm in update_alarms:
+                if alarm not in valid_alarms:
+                    verrors.add(
+                        'netdata_update.alarms',
+                        f'{alarm} not a valid alarm'
+                    )
+                else:
+                    if not isinstance(alarm, str):
+                        verrors.add(
+                            'netdata_update.alarms',
+                            f'{alarm} key must be a string'
+                        )
+                    if not isinstance(update_alarms[alarm], bool):
+                        verrors.add(
+                            'netdata_update.alarms',
+                            f'{alarm} value can only be boolean'
+                        )
+
+        if verrors:
+            raise verrors
+
+        # TODO: See what can be done to improve this section - is very crude right now - We are probably not getting
+        # templates right now, look into that
+        for alarm in valid_alarms:
+            if alarm not in data['alarms']:
+                data['alarms'][alarm] = True
+
+        for alarm in update_alarms:
+            # These are valid alarms
+            data['alarms'][alarm] = update_alarms[alarm]
+
+        data['memory_mode'] = data['memory_mode'].lower()
+
+        return data
+
+    @accepts(
+        Dict(
+            'netdata_update',
+            Str('additional_params'),
+            Dict(
+                'alarms',
+                additional_attrs=True
+            ),
+            IPAddr('bind_to'),  # TODO: nginx.conf will need to be adjusted accordingly
+            Int('bind_to_port', validators=[Port()]),
+            Int('history'),
+            Int('http_port_listen_backlog'),
+            Str('memory_mode', enum=['SAVE', 'MAP', 'RAM']),
+            Int('update_every')
+        )
+    )
+    async def do_update(self, data):
+        old = await self.config()
+        new = old.copy()
+        # We separate alarms we have in db and the ones user supplies. If alarms are valid, we add them to db, else
+        # we keep the old ones
+        new['update_alarms'] = data.pop('alarms', {})
+        new.update(data)
+
+        new = await self.validate_attrs(new)
+
+        await self._update_service(old, new)
+
+        return await self.config()
+
+
+class NetDataStreamingMetrics(SystemServiceService):
```

Comment:
Why have two services?

That means in UI the user will have to save two different screens?